js/import/util/metadata.js

```diff
@@ -0,0 +1,169 @@
+import alert from '../../shared/alert.js';
+import { saveImporterSectionsMapping } from '../../sections-mapping/utils.ui.js';
+import { createElement } from '../../shared/utils.js';
+
+const METADATA_EDITOR = document.getElementById('metadata-editor');
+const METADATA_EDITOR_SECTIONS = document.getElementById('metadata-editor-rows');
+const ADD_METADATA_BUTTON = document.getElementById('metadata-editor-add');
+
+// Handler for deleting a row for this scope (copied until `mappingData` is centralized).
+let getMetadataRowDeleteButton;
+
+/**
+ * Either find the matching mapping and update it, create a brand new one or indicate that there
+ * is a duplicate.
+ * @param event
+ * @param mappingData
+ * @param originalURL
+ */
+const updateMetadataMapping = (event, mappingData, originalURL) => {
+  if (event.target.value !== undefined) {
+    const metadataFields = ['name', 'value', 'condition'];
+    const row = event.target.parentElement;
+    const id = row.dataset.metadataId;
+    const mItem = mappingData.find((m) => m.id === id) ?? {};
+    metadataFields.forEach((attr) => {
```

Comment:
Now only one variable `mItem` which is still checked for duplicates before applying below.